File: scripts/get_install_steps.ts
// FILE: scripts/get_install_steps.ts
import { readFileSync } from 'fs';
import { join } from 'path';
import { parse } from 'csv-parse/sync';

interface InstallationStep {
  software: string;
  os: string;
  installation_steps: string;
  source: string;
}

let installationSteps: InstallationStep[] = [];

export function loadInstallationSteps(): void {
  try {
    const csvPath = join(process.cwd(), 'data', 'installation_steps.csv');
    const fileContent = readFileSync(csvPath, 'utf-8');
    const records = parse(fileContent, {
      columns: true,
      skip_empty_lines: true,
      bom: true
    });
    installationSteps = records as InstallationStep[];
  } catch (error) {
    console.error('Error loading installation steps:', error);
    installationSteps = [];
  }
}

export function getInstallationSteps(software: string, os: string): string[] | null {
  if (installationSteps.length === 0) {
    loadInstallationSteps();
  }
  
  const normalizedSoftware = software.trim();
  const normalizedOS = os.trim();
  
  const match = installationSteps.find(
    step => 
      step.software.toLowerCase() === normalizedSoftware.toLowerCase() &&
      step.os.toLowerCase() === normalizedOS.toLowerCase()
  );
  
  if (!match) {
    return null;
  }
  
  return match.installation_steps.split(' || ').filter(step => step.trim().length > 0);
}

File: models/get_install_steps.py
# FILE: models/get_install_steps.py
import pandas as pd
import sys
import os
import json

def load_installation_steps():
    """Load installation steps from CSV"""
    csv_path = os.path.join(os.path.dirname(__file__), '..', 'data', 'installation_steps.csv')
    df = pd.read_csv(csv_path)
    return df

def get_installation_steps(software, os_name):
    """Get installation steps for software and OS"""
    df = load_installation_steps()
    
    normalized_software = software.strip().lower()
    normalized_os = os_name.strip().lower()
    
    match = df[
        (df['software'].str.lower() == normalized_software) &
        (df['os'].str.lower() == normalized_os)
    ]
    
    if match.empty:
        return None
    
    steps_str = match.iloc[0]['installation_steps']
    steps = [step.strip() for step in steps_str.split(' || ') if step.strip()]
    return steps

if __name__ == '__main__':
    # Command line interface
    if len(sys.argv) < 3:
        print("Usage: python get_install_steps.py <software> <os>")
        sys.exit(1)
    
    software = sys.argv[1]
    os_name = sys.argv[2]
    result = get_installation_steps(software, os_name)
    
    if result:
        print(json.dumps(result))
    else:
        print("null")

